applications/HDF5Application/custom_utilities/factor_elements_and_conditions_utility.cpp

```diff
@@ -0,0 +1,274 @@
+// System includes
+#include <sstream>
+
+// External includes
+#ifdef KRATOS_USING_MPI
+#include "mpi.h"
+#endif
+
+// Project includes
+#include "utilities/compare_elements_and_conditions_utility.h"
+
+// Application includes
+#include "custom_utilities/factor_elements_and_conditions_utility.h"
+
+namespace Kratos
+{
+
+template<class TContainerType>
+// ElementContainerType || ConditionContainerType
+std::vector<std::string> GetLocalComponentNames(const TContainerType& rElements)
+{
+    KRATOS_TRY;
+
+    if (rElements.size() == 0)
+        return std::vector<std::string>();
+
+    std::vector<std::pair<std::string, const typename TContainerType::data_type *>> components(1);
+    std::string name;
+    // Add first component.
+    CompareElementsAndConditionsUtility::GetRegisteredName(rElements.front(), name);
+    components[0] = std::pair<std::string, const typename TContainerType::data_type *>(
+        name, &rElements.front());
+
+    // Find all remaining components.
+    components.reserve(5);
+    unsigned pos = 0;
+    for (typename TContainerType::const_iterator it = rElements.begin() + 1;
+         it != rElements.end(); ++it)
+    {
+        if (GeometricalObject::IsSame(*it, *components[pos].second) == false)
+        {
+            bool found = false;
+            for (unsigned k = 0; k < components.size(); ++k)
+            {
+                if (GeometricalObject::IsSame(*it, *components[k].second))
+                {
+                    found = true;
+                    pos = k;
+                    break;
+                }
+            }
+
+            if (!found)
+            {
+                CompareElementsAndConditionsUtility::GetRegisteredName(*it, name);
+                components.push_back(std::pair<std::string, const typename TContainerType::data_type*>(
+                    name, &(*it)));
+                pos = components.size() - 1;
+            }
+        }
+    }
+
+    // Return component names.
+    std::vector<std::string> component_names(components.size());
+    for (unsigned i = 0; i < components.size(); ++i)
+        component_names[i] = components[i].first;
+    return component_names;
+
+    KRATOS_CATCH("");
+}
+
+template<class TContainerType>
+// ElementContainerType || ConditionContainerType
+std::vector<std::string> GetComponentNames(const TContainerType& rElements)
+{
+    KRATOS_TRY;
+
+    const int max_size = 1000;
```

Comment:
constexpr?

Reply:
I think either is fine here but will use constexpr.